common/smoke-test/Update-SmokeTests.ps1

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env pwsh
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+
+param (
+  [Parameter(Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $TestApplicationId,
+
+  [Parameter()]
+  [string] $TestApplicationSecret,
+
+  [Parameter()]
+  [string] $TestApplicationOid,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string] $TenantId,
+
+  [Parameter(ParameterSetName = 'Provisioner')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $SubscriptionId,
+
+  [Parameter()]
+  [string] $Location = '',
+
+  [Parameter()]
+  [string] $Environment = 'AzureCloud',
+
+  [Parameter()]
+  [hashtable] $AdditionalParameters,
+
+  [Parameter()]
+  [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID),
+
+  # Captures any arguments not declared here (prevents no parameter errors)
+  [Parameter(ValueFromRemainingArguments = $true)]
+  $RemainingArguments
+)
+function SetEnvironmentVariable {
+  param(
+    [string] $Name,
+    [string] $Value
+  )
+
+  if ($CI) {
+    Write-Host "##vso[task.setvariable variable=_$Name;issecret=true;]$($Value)"
+    Write-Host "##vso[task.setvariable variable=$Name;]$($Value)"
+  }
+  else {
+    Write-Verbose "Setting local environment variable: $Name = ***"
+    Set-Item -Path "env:$Name" -Value $Value
+  }
+}
+
+pushd ../../
+
+Write-Verbose "Reading manifest..."
+$manifest = (Get-Content -Path common/smoke-test/samples-manifest.json | ConvertFrom-Json)
```

Comment:
Can we generate this list without needing to have an explicit manifest checked in.